gravityflow/gravityflowstripe
# Update the wait for subscription event to support subscriptions started by other entries

re: [HS#14266](https://secure.helpscout.net/conversation/1238123523/14266?folderId=793490)

This adds the Subscription ID Field setting to the Wait for Subscription Event step so the step can be used to pause a workflow pending events for a subscription created by another entry.

`Gravity_Flow_Stripe::action_gform_post_payment_callback()` has also been updated to support this change.

In the above customers case they want to pause the workflow once the Cancel Subscription step has run and wait until the end of the billing period when the subscription is cancelled for the next step to run.

## Testing Instructions
- For this you'll need a live site configured to receive webhooks or the Stripe CLI configured to listen and forward webhooks to your local test site
- Create and submit a form to start a Stripe Subscription. The payment collection method and billing cycle can be whatever you want.
- Import and preview form: https://www.dropbox.com/s/nzll0x5nivlxebf/gravityforms-export-2020-08-15.json?dl=0
- Submit with the subscription id from the first submission
- Go to workflow status and confirm the workflow is on the wait step
- Go to https://dashboard.stripe.com/test/subscriptions/the_sub_id_here
- Click the ... menu and select the cancel now option: https://d.pr/i/pJ8SNG
- Return to workflow status to confirm the workflow was resumed and is now on the approval step
